backend/src/transaction_event/protocol_level_tokens.rs

```diff
@@ -714,4 +998,70 @@ impl PreparedTokenUpdate {
         .await?;
         Ok(())
     }
+
+    async fn update_account_balance_and_check_zero(
+        &self,
+        tx: &mut sqlx::PgTransaction<'_>,
+        account: &str,
+        amount: &BigDecimal,
+    ) -> anyhow::Result<BigDecimal> {
+        let final_balance = sqlx::query_scalar!(
+            "INSERT INTO plt_accounts (account_index, token_index, amount, decimal)
+             VALUES ((SELECT index FROM accounts WHERE address = $1), (SELECT index FROM \
+             plt_tokens WHERE token_id = $2), $3, (SELECT decimal FROM plt_tokens WHERE token_id \
+             = $2))
+             ON CONFLICT (account_index, token_index) DO UPDATE SET amount = plt_accounts.amount + \
+             $3
+             RETURNING amount",
+            account,
+            self.token_id,
+            amount
+        )
+        .fetch_optional(tx.as_mut())
+        .await?
+        .flatten()
+        .unwrap_or_else(|| BigDecimal::from(0));
+
+        Ok(final_balance)
+    }
+
+    async fn update_metrics_plt_unique_account_count_decrement(
```

Comment:
I would keep the two functions `update_metrics_plt_unique_account_count_decrement` and `update_metrics_plt_unique_account_count_increment` together.